Source/ConstraintMakerPriortizable.swift

```diff
@@ -30,38 +30,49 @@
 
 public class ConstraintMakerPriortizable: ConstraintMakerFinalizable {
     
-    @discardableResult
-    public func priority(_ amount: ConstraintPriorityTarget) -> ConstraintMakerFinalizable {
-        self.description.priority = amount
-        return self
-    }
-    
-    @available(*, deprecated:3.0, message:"Use priority(_ amount: ConstraintPriorityTarget) instead.")
-    @discardableResult
-    public func priorityRequired() -> ConstraintMakerFinalizable {
-        return self.priority(1000)
+    public enum Priority: Strideable, RawRepresentable, Equatable {
+        case low
+        case medium
+        case high
+        case required
+        case custom(value: Float)
+        
+        public init?(rawValue: Float) {
+            // TODO:
+            // constrain to 0 <= x <= 1000
```

Comment:
since it's a failable initializer we should return nil for invalid priority (not completely sure that <0 or >1000 is invalid...)